### ISSUE TITLE
[BUG] Unable to resolve reference for another package when updating shrinkwrap

### ISSUE THREAD

**Issue by Artur-** [npm/cli#4409]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

When an existing shrinkwrap is installed
```
npm init
npm i @polymer/polymer@3.2.0 @vaadin/vaadin-core-shrinkwrap@23.0.0-beta1 --save
```

And you lock the version of a package not mentioned in the shrinkwrap
```
# Add "overrides": { "@polymer/polymer": "$@polymer/polymer"} to package.json
```

You can then no longer upgrade the shrinkwrap
```
npm i @vaadin/vaadin-core-shrinkwrap@23.0.0-beta2 
```

```
npm ERR! Unable to resolve reference $@polymer/polymer
```

### Expected Behavior

You can install the new shrinkwrap even if an unrelated package has an `override` defined.

### Steps To Reproduce

1. git clone https://github.com/Artur-/npm-overrides-test.git
2. cd npm-overrides-test
3. npm i @vaadin/vaadin-core-shrinkwrap@23.0.0-beta2 

### Environment

- npm: 8.5.0
- Node.js:v17.3.0
- OS Name: Mac os Big Sur
- System Model Name: Macbook Pro
- npm config:
```ini
//registry.npmjs.org/:_authToken = (protected) 
```

**Comment by Artur-** [comment]
Sounds a bit similar to #4395

**Comment by Artur-** [comment]
Debug log says
```
38 verbose stack Error: Unable to resolve reference $@polymer/polymer
38 verbose stack     at Edge.get spec [as spec] (/Users/artur/.nvm/versions/node/v17.3.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/edge.js:177:15)
38 verbose stack     at Edge.satisfiedBy (/Users/artur/.nvm/versions/node/v17.3.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/edge.js:95:32)
38 verbose stack     at Edge.[_loadError] (/Users/artur/.nvm/versions/node/v17.3.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/edge.js:212:15)
38 verbose stack     at Edge.reload (/Users/artur/.nvm/versions/node/v17.3.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/edge.js:224:38)
38 verbose stack     at Node.[_reloadNamedEdges] (/Users/artur/.nvm/versions/node/v17.3.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/node.js:1348:12)
38 verbose stack     at Node.[_reloadNamedEdges] (/Users/artur/.nvm/versions/node/v17.3.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/node.js:1351:27)
38 verbose stack     at Node.set root [as root] (/Users/artur/.nvm/versions/node/v17.3.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/node.js:634:40)
38 verbose stack     at Node.set parent [as parent] (/Users/artur/.nvm/versions/node/v17.3.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/node.js:1235:15)
38 verbose stack     at Node.replace (/Users/artur/.nvm/versions/node/v17.3.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/node.js:1147:20)
38 verbose stack     at PlaceDep.replaceOldDep (/Users/artur/.nvm/versions/node/v17.3.0/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/place-dep.js:426:17)
```

Looking at the code in question 
```
        const ref = this.overrides.value.slice(1)
        const pkg = this.from.root.package
```

reveals that `ref` refers to `@polymer/polymer` but `pkg` i.e. `this.from.root.package` refers to the shrinkwrap package.json, not the project package.json as expected

**Comment by fritzy** [comment]
Closing this as a duplicate for now for #4395, but this specifically calls out shrinkwraps. If it turns out to be a separate issue, we'll reopen.

**Comment by Artur-** [comment]
This has been resolved by #4395 and npm 8.16.0, thanks
